internal/tools/aws/database_multitenant.go

```diff
@@ -0,0 +1,637 @@
+	defer cancel()
+
+	err = d.createDatabaseIfNotExist(ctx, databaseName)
+	if err != nil {
+		return errors.Wrapf(err, "unable to create schema in RDS cluster ID %s", *lockedCluster.cluster.DBClusterIdentifier)
+	}
+
+	installationSecreteName := RDSMultitenantSecretName(d.installationID)
+
+	installationSecretValue, err := d.client.Service().secretsManager.GetSecretValue(&secretsmanager.GetSecretValueInput{
+		SecretId: aws.String(installationSecreteName),
+	})
+	if err != nil && !IsErrorCode(err, secretsmanager.ErrCodeResourceNotFoundException) {
+		return errors.Wrapf(err, "failed to get RDS database secret for installation ID %s", d.installationID)
+	}
+
+	var installationSecret *RDSSecret
+	if installationSecretValue != nil && installationSecretValue.SecretString != nil {
+		installationSecret, err = unmarshalSecretPayload(*installationSecretValue.SecretString)
+		if err != nil {
+			return errors.Wrapf(err, "failed to unmarshal RDS database secret for installation ID %s", d.installationID)
+		}
+	} else {
+		description := RDSMultitenantClusterSecretDescription(d.installationID, *lockedCluster.cluster.DBClusterIdentifier)
+		tags := []*secretsmanager.Tag{
+			{
+				Key:   aws.String(trimTagPrefix(DefaultRDSMultitenantDBClusterIDTagKey)),
+				Value: lockedCluster.cluster.DBClusterIdentifier,
+			},
+			{
+				Key:   aws.String(trimTagPrefix(DefaultRDSMultitenantVPCIDTagKey)),
+				Value: aws.String(*vpc.VpcId),
+			},
+			{
+				Key:   aws.String(trimTagPrefix(DefaultMattermostInstallationIDTagKey)),
+				Value: aws.String(d.installationID),
+			},
+		}
+		installationSecret, err = d.createInstallationSecret(installationSecreteName, d.installationID, description, tags)
+		if err != nil {
+			return errors.Wrapf(err, "failed to create a RDS database secret for installation ID %s", d.installationID)
+		}
+	}
+
+	err = d.createUserIfNotExist(ctx, installationSecret.MasterUsername, installationSecret.MasterPassword)
+	if err != nil {
+		return errors.Wrapf(err, "failed to create a Mattermost database schema for installation ID %s", d.installationID)
+	}
+
+	err = d.grantUserFullPermissions(ctx, databaseName, installationSecret.MasterUsername)
+	if err != nil {
+		return errors.Wrapf(err, "failed to grant permissions to Mattermost database user for installation ID %s", d.installationID)
+	}
+
+	databaseCluster, err := store.GetDatabaseCluster(*lockedCluster.cluster.DBClusterIdentifier)
+	if err != nil {
+		return errors.Wrapf(err, "failed to get database cluster for RDS cluster ID %s", *lockedCluster.cluster.DBClusterIdentifier)
+	}
+
+	databaseClusterInstallations, err := databaseCluster.GetInstallations()
+	if err != nil {
+		return errors.Wrapf(err, "failed to get database cluster installations for RDS cluster ID %s", *lockedCluster.cluster.DBClusterIdentifier)
+	}
+
+	if !databaseClusterInstallations.Contains(d.installationID) {
```

Comment:
I am a little confused by this. In the beginning of this method we determine that a databaseCluster has this Installation in it as that is how the RDS cluster is determined. There really should be no way for this if statement to be true, right?

Reply:
At the beginning of this method we create a database name, get the claimed VPC, acquire a lock for some RDS cluster if any, and then create the database. The condition here is just in case of the update counter fails we don't try to update the store again.

Reply:
It got resolved by creating specific datastore methods for adding and removing installations.